### ISSUE TITLE
host maintenance mode

### ISSUE THREAD

**Issue by hickeng** [vmware/vic#860]
This is primarily a testing exercise as the implementation is dependent on #770. This is pulled out as a separate issue because of how important it is that we not inhibit this pervasive day2 operation.

This test asserts that a host, with containerVMs on it, can be put into maintenance mode without manual intervention. There *will* be a case where this is problematic - the case where there is an attach operation in progress to a containerVM on that host.

There is one implementation component to it that may or may not be viable:
* if there is an attach session open to a containerVM on the quiescing host and
* the applianceVM can receive a notification about the hosts change of status
  * then we can consider dropping the attach connection to allow the containerVM to vMotion.

This is getting very close to the behaviour implemented via the vSPC and if we wish to support this we should seriously consider doing so leveraging that mechanism (see the [communications document](https://github.com/vmware/vic/blob/master/doc/design/communications.md#vspc-relay) for vSPC details - there is scope for implementing it directly in the attach server rather than as an agent).

**Comment by mhagen-vmware** [comment]
See #2245 
See #2244
